### PR TITLE
Feature/aliases table simple experiment

### PR DESCRIPTION
**Pull request by danoswaltCL**
This change creates the MVP for the editable condition alias table in the UI. This can be merged in before backend changes (submitting experiment will continue to work, the backend just won't be saving the alias details yet).

### CODE REVIEW

File: frontend/projects/upgrade/src/app/core/experiments/experiment-utility.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { ExperimentPartition, ExperimentCondition } from './store/experiments.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExperimentUtilityService {
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/experiments/experiment-utility.service.ts:7]: I created an additional service to off-load some of the business logic that is clogging up the component classes that doesn't really need to be there. This is especially true for validation logic and other functions that are not directly related to view-model logic.

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts [hunk 2]
```diff
@@ -228,8 +248,15 @@ export interface ExperimentState extends EntityState<Experiment> {
   allExperimentNames: {};
   contextMetaData: {};
   updatedStat?: IExperimentEnrollmentDetailStats;
+  isAliasTableEditMode: boolean;
+  aliasTableEditIndex: number | null;
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:252]: I let this be able to be `null` when `isAliasTableEditMode` is false, as there is no row-index being edited at that time.

**Comment by danoswaltCL** [comment]
found a bug when testing integration, disabled merge for now...

**Comment by ppratikcr7** [comment]
> found a bug when testing integration, disabled merge for now...

What's the bug? Let us know, we will resolve it.

**Comment by danoswaltCL** [comment]
@ppratikcr7 ...found UI bug, I should say, but I was also seeing backend issue saying "violating Foreign Key constraint" when posting the alias conditions.... couldn't confirm if it was issue on my end or if is real backend error

**Comment by danoswaltCL** [comment]
ok this is good to go whenever now, @RidhamShah @ppratikcr7 integrating with the `fix/issue-457` branch looks like it's working as expected 👍

**Comment by danoswaltCL** [comment]
branch has been updated to now complete adding context metadata filtered combo-box to the alias input rows when editing. FYI this change added some changes to context metadata type models so that they actually gave accurate type information; I regression tested context-metadata behavior in several places, hopefully nothing was affected.

**Comment by ppratikcr7** [comment]
> @ppratikcr7 ...found UI bug, I should say, but I was also seeing backend issue saying "violating Foreign Key constraint" when posting the alias conditions.... couldn't confirm if it was issue on my end or if is real backend error

Hope this is sorted now. :)

**Comment by danoswaltCL** [comment]
I have a couple changes I need to push up now after seeing finalized integration code, hold up on merging! backend code looks good to me

**Comment by danoswaltCL** [comment]
alright this is updated and ready to go, please review

**Review by jreddig — APPROVED** [review]
looks lovely!
